### DISCUSSION TITLE
Define shortcut like actions.

### DISCUSSION THREAD

**Discussion by shodan65** [Tasshack/dreame-vacuum#672] [Ideas]
I would like to be able to define actions that are like the shortcuts defined in ther Dreame App, but more capable.
I have a multilayer map and a x40. My base is located on the ground floor. The robot is located in the base.

The action to clean the 1st floor should do something like this:

- Refill the watertank and pickup the mop
- Move the robot to a predefined spot on the map
- Wait to be picked up and be moved to the 1st floor
After being placed on the 1st floor and user response
- change the map to 1st-floor
- perform the cleaning sequence
- return to the entry point and wait to be picked up
After being placed on the ground floor and user response
- switch the map
- return to the base

Any ideas on how to make that happen?

**Comment by Tasshack** [comment]
https://www.home-assistant.io/integrations/script/

**Reply by shodan65** [reply]
Got your point. I'm new to the dreame world and I was looking for a practical example. That solution would be somehow a script is obvious.

So how would a script handle the 1st two task? How could a script wait for user response?

**Reply by Tasshack** [reply]
You should start with eliminating the actions cannot be done by the app like `Refill the watertank and pickup the mop`. 
Then you should perform the actions manually from the app and watch the HA logbook events to see which entity values are changed to what.
Finally you should check the [services](https://github.com/Tasshack/dreame-vacuum/blob/dev/docs/services.md) and [entities](https://github.com/Tasshack/dreame-vacuum/blob/dev/docs/entities.md) docs and find out which services do you need to call on your script. You will be needing [entity services](https://www.home-assistant.io/integrations/select/#action-selectselect_option) for modifying their values so should should also checkout that. 

You need to all this work yourself and cannot expect anyone to write a script for you to just copy and paste.

**Reply by shodan65** [reply]
I don't want anybody to write my scripts. I just wanted to discuss an idea on how to utilize your extension in a specific way.

**Reply by shodan65** [reply]
And maybe get some advice.

**Reply by Tasshack** [reply]
You cannot find that kind of information from anybody in this repo. Nobody helps to no one here becausen its not a forum or a customer community. Try reddit!
